sdk/src/main/java/com/horizen/account/transaction/EthereumTransaction.java

```diff
@@ -115,16 +115,52 @@ public TransactionSerializer serializer() {
 
     @Override
     public void semanticValidity() throws TransactionSemanticValidityException {
+        if (getValue().signum() < 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "negative value", id()));
+        if (getNonce().signum() < 0)
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "negative nonce", id()));
         if (getGasLimit().signum() <= 0)
-            throw new TransactionSemanticValidityException("Cannot create transaction with zero gas limit");
-        if (this.isSigned()) {
-            if (this.getFrom().address().length != Account.ADDRESS_SIZE)
-                throw new TransactionSemanticValidityException("Cannot create signed transaction without valid from address");
-            if (!this.getSignature().isValid(this.getFrom(), this.messageToSign()))
-                throw new TransactionSemanticValidityException("Cannot create signed transaction with invalid " +
-                        "signature");
-
+            throw new TransactionSemanticValidityException(String.format("Transaction [%s] is semantically invalid: " +
+                    "non-positive gas limit", id()));
+        if (getTo() == null && getData().length == 0)
```

Comment:
good point.
`getTo` expect to return `null` in case of smart contract declaration.
And now we probably have a bug in our EthTransaction implementation. Must return `null` as soon as `getToAddress` return null